In [ ]:
import numpy as np
import pandas as pd

In [2]:
# load drinks dataset from csv files:
drinks = pd.read_csv('../datasets/drinks.csv')


In [3]:
# get the shape:
drinks.shape

(193, 6)

From the shape, we can see that there are 193 rows and 6 columns

In [4]:
# show first 3 rows from drinks dataframe uing the head method:
drinks.head(3)

country  beer_servings  spirit_servings  wine_servings  \
0  Afghanistan              0                0              0   
1      Albania             89              132             54   
2      Algeria             25                0             14   

   total_litres_of_pure_alcohol continent  
0                           0.0        AS  
1                           4.9        EU  
2                           0.7        AF

In [5]:
drinks.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 193 entries, 0 to 192
Data columns (total 6 columns):
country                         193 non-null object
beer_servings                   193 non-null int64
spirit_servings                 193 non-null int64
wine_servings                   193 non-null int64
total_litres_of_pure_alcohol    193 non-null float64
continent                       170 non-null object
dtypes: float64(1), int64(3), object(2)
memory usage: 9.2+ KB


In [6]:
# show rows with NaN values (i.e. empty cells):
nan_mask = drinks.continent.isnull()

drinks[nan_mask].head(3)

country  beer_servings  spirit_servings  wine_servings  \
5   Antigua & Barbuda            102              128             45   
11            Bahamas            122              176             51   
14           Barbados            143              173             36   

    total_litres_of_pure_alcohol continent  
5                            4.9       NaN  
11                           6.3       NaN  
14                           6.3       NaN

In [7]:
# show only country names without continent values
drinks.country[nan_mask].head(3)

5     Antigua & Barbuda
11              Bahamas
14             Barbados
Name: country, dtype: object

In [8]:
# get some stats:
drinks.describe()

beer_servings  spirit_servings  wine_servings  \
count     193.000000       193.000000     193.000000   
mean      106.160622        80.994819      49.450777   
std       101.143103        88.284312      79.697598   
min         0.000000         0.000000       0.000000   
25%        20.000000         4.000000       1.000000   
50%        76.000000        56.000000       8.000000   
75%       188.000000       128.000000      59.000000   
max       376.000000       438.000000     370.000000   

       total_litres_of_pure_alcohol  
count                    193.000000  
mean                       4.717098  
std                        3.773298  
min                        0.000000  
25%                        1.300000  
50%                        4.200000  
75%                        7.200000  
max                       14.400000

In [9]:
# get the maximum wine_servings value:

max_wine = drinks.wine_servings.max()
print( f'max_wine value = {max_wine}' )

# make the mask
max_wine_mask = drinks.wine_servings == max_wine

# get the value of max_wine drinking country:
max_wine_country = drinks.country[max_wine_mask].values[0]

print( f'Country drinking max wine: {max_wine_country}' )

max_wine value = 370
Country drinking max wine: France


In [10]:
# get the maximum beer_servings country:
max_beer = drinks.beer_servings.max()
print( f'max_beer value = {max_beer}' )

# get the value of max_beer drinking country:
max_beer_country = drinks[drinks.beer_servings == max_beer].values[0,0]

print( f'Country drinking max beer: {max_beer_country}' )

max_beer value = 376
Country drinking max beer: Namibia


In [11]:
# get the row for maximum spirit_servings:
max_spirit = drinks.spirit_servings.max()
drinks[drinks.spirit_servings == max_spirit]

country  beer_servings  spirit_servings  wine_servings  \
68  Grenada            199              438             28   

    total_litres_of_pure_alcohol continent  
68                          11.9       NaN

In [12]:
## get the maximum spirit_servings for each continent:
drinks.groupby(by='continent').spirit_servings.max()

continent
AF    152
AS    326
EU    373
OC    254
SA    302
Name: spirit_servings, dtype: int64

In [13]:
# list only countries which name startswith 'a'
for country in drinks.country:
    if country[0].lower() == 'a':
        print(country)

Afghanistan
Albania
Algeria
Andorra
Angola
Antigua & Barbuda
Argentina
Armenia
Australia
Austria
Azerbaijan


In [14]:
# print the row with Bulgaria:
drinks[ drinks.country == 'Bulgaria'].wine_servings

# print the wine-servings in Bulgaria:
bul_wine_servings = drinks[ drinks.country == 'Bulgaria'].wine_servings.values[0]
print(bul_wine_servings)

94


In [15]:
# TASK: which countries drinks more spirit than Bulgaria
